### PR TITLE
CAF-3538: Add Document Worker and Composite Document Worker policy ty…

### PR DESCRIPTION
**Pull request by mmaBel**
https://jira.autonomy.com/browse/CAF-3538

…pes to database base data so they are available for chained worker logic.

### CODE REVIEW

File: Database/src/main/resources/changelog1.xml [hunk 1]
```diff
@@ -781,4 +781,36 @@
...
...:&quot;Composite Document Policy Type&quot;,&quot;description&quot;:&quot;Allows the Policy Worker to integrate with workers which meet the Document Worker contract.  Document Workers can add extra fields to a document, and can also update or remove fields.&quot;,&quot;type&quot;:&quot;object&quot;,&quot;properties&quot;:{&quot;workerName&quot;:{&quot;description&quot;:&quot;The name of the worker. Some configuration details, such as the default queue names, will be looked up based on this name.&quot;,&quot;type&quot;:&quot;string&quot;},&quot;queueName&quot;:{&quot;description&quot;:&quot;Optional. Typically the target queue name is provided via configuration. This field can be used as a means to override the queue name specified in the configuration.&quot;,&quot;type&quot;:&quot;string&quot;},&quot;diagnosticsQueueName&quot;:{&quot;description&quot;:&quot;Optional. Allows a second 'diagnostic' queue to be specified for the worker. This will be used if the worker fails.&quot;,&quot;type&quot;:&quot;string&quot;}},&quot;customData&quot;:{&quot;description&quot;:&quot;Optional. Used to send additional worker-specific data to the worker.&quot;,&quot;type&quot;:&quot;object&quot;}}"/>
```
Comment (dermot-hardy) [inline Database/src/main/resources/changelog1.xml]: Should `customData` not be inside `properties`?

File: Database/src/main/resources/changelog1.xml [hunk 2]
```diff
@@ -781,4 +781,36 @@
         </insert>
     </changeSet>
 
+    <changeSet author="CAF" id="1512411132000-1">
+        <insert tableName="tbl_policy_type">
+            <column name="item_name" value="Document Policy Type"/>
```
Comment (dermot-hardy) [inline Database/src/main/resources/changelog1.xml]: I don't know if we really want to add this as it doesn't support chaining.

**Comment by mmaBel** [comment]
This needs updated to specify an entry for type 'ChainedAction', using same settings as CompositeDocumentWorkerHandler. Remove entries for Composite and DocumentWorkerHandler.

File: Database/src/main/resources/changelog1.xml (hunk 1 above)
Comment (mmaBel) [reply Database/src/main/resources/changelog1.xml]: It should, looks like an error in the policy definition this was copied from. The schema only limits those properties it knows about so no validation occurs on customData during creation.

File: Database/src/main/resources/changelog1.xml (hunk 2 above)
Comment (mmaBel) [reply Database/src/main/resources/changelog1.xml]: Removed. Composite Document type also removed in favor of Chained Action.

File: release-notes-1.1.0.md [hunk 3]
```diff
@@ -6,4 +6,7 @@ ${version-number}
 #### New Features
 - Use latest release of policy-elasticsearch-container base image with updated copyright and logging.
 
+- [CAF-3538](https://jira.autonomy.com/browse/CAF-3538): Add Chained Action policy type to database base data.
+  The Chained Action policy type has been added to the base data of the database installer. This is is so that it is available by default in a core policy system for use with the chained workers enhancement. Previously a policy worker would have been responsible for registering types in the database but that worker is not present when using chaining. The new Chained Action type supports the same properties as Composite Document Worker and is intended to replace it for use in chaining of workers going forward.
```
Comment (dermot-hardy) [inline release-notes-1.1.0.md]: The text is fine but you need to leave two spaces after the header so that it doesn't run directly into the text.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**